### PR TITLE
Add support for vLLM via openai plugin.

### PR DESCRIPTION
**Pull request by dagrayvid**
This PR combines changes suggested in https://github.com/openshift-psap/llm-load-test/pull/39 and https://github.com/openshift-psap/llm-load-test/pull/27, in order to support load testing models running in vLLM via the OpenAI conformant REST API.

### CODE REVIEW

File: plugins/openai_plugin.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+from result import RequestResult
+
+urllib3.disable_warnings()
+"""
+Example plugin config.yaml:
+
+plugin: "openai_plugin"
+plugin_options:
+  streaming: True/False
+  host: "http://127.0.0.1:5000/v1/completions"
```
Comment (ccamacho) [inline plugins/openai_plugin.py:18]: what about  v1/chat/completions ?

File: plugins/openai_plugin.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+
+    def streaming_request_http(self, query: dict, user_id: int, test_end_time: float):
+        headers = {"Content-Type": "application/json"}
+
+        data = {
+            "prompt": query["text"],
+            "max_tokens": query["output_tokens"],
+            "min_tokens": query["output_tokens"],
+            "temperature": 1.0,
```
Comment (ccamacho) [inline plugins/openai_plugin.py:121]: Does these options need to be hardwired? Also there is the option to do queries without the stream option, on the other hand we only use "stream" right? Across all the plugins?

**Review by ccamacho — APPROVED** [review]

File: plugins/openai_plugin.py (hunk 1 above)
Comment (dagrayvid) [reply plugins/openai_plugin.py:18]: Hmm, maybe I should expose this as an option, so we can hit the chat endpoint or the completions endpoint.

File: plugins/openai_plugin.py (hunk 2 above)
Comment (dagrayvid) [reply plugins/openai_plugin.py:121]: In some cases we may want to test stream and non-streaming. 

I think we should keep max/min tokens set to make sure we are getting consistent numbers that don't vary based on when the model predicts the stop token.

As for temperature, top_p, and seed... no those don't need to be hardcoded, and maybe should be configurable, but I think this is low priority so will leave it to a follow-up PR.

**Pull request merged**